src/test/debuginfo/no_mangle-info.rs

```diff
@@ -0,0 +1,26 @@
+// compile-flags:-g
+
+// === GDB TESTS ===================================================================================
+
+// gdb-command:run
+// gdb-command:whatis TEST
+// gdb-check:type = u64
+
+// === LLDB TESTS ==================================================================================
+
+// lldb-command:run
+// lldb-command:expr TEST
+// lldb-check: (unsigned long) $0 = 3735928559
+
+// === CDB TESTS ==================================================================================
+// cdb-command: g
+
+// cdb-command: dx a!no_mangle_info::TEST
+// cdb-check: a!no_mangle_info::TEST : 0xdeadbeef [Type: unsigned __int64]
```

Comment:
That is surprising! I tried what I believe is equivalent C++ and the namespace does not appear to be included:

```c++
#include <iostream>
using namespace std;

namespace ThisIsMyNamespace {
    extern "C" static int SomeInteger = 42;
}

int main()
{
    cout << ThisIsMyNamespace::SomeInteger;
}
```

In windbg:

```
0:000> x cpptest!*SomeInteger
0064a000          cpptest!SomeInteger = 0n42
```